metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1203,6 +1210,23 @@ def gen_schema_metadata(
     def get_report(self) -> BigQueryV2Report:
         return self.report
 
+    def get_views_for_dataset(
+        self, conn: bigquery.Client, project_id: str, dataset_name: str
+    ) -> Iterable[BigqueryView]:
+
+        with PerfTimer() as timer:
+            table_items = self.get_core_table_details(conn, dataset_name, project_id)
```

Comment:
I see, this is additional bigquery api call added in this PR. While functionally it works well. I'm slightly concerned about api call overhead as we already make this call in `self.get_tables_for_dataset` for same dataset and project. Ideally we should be able to reuse the old results using lru_cache(maxsize=1). Also, this call inside `get_views_for_dataset` should be skipped if capture_table_label_as_tag=False. table_items will be `Optional[Dict[str, TableListItem]]`.

For the record, labels could also be fetched by updating existing `views_for_dataset...` query used in `BigQueryDataDictionary.get_views_for_dataset` to join with INFORMATION_SCHEMA.TABLE_OPTIONS where OPTION_NAME = "labels". We already have similar query to fetch view description. However its not clear if this option is better/worse than the current implementation so I'm fine with the way it is done right now..

Reply:
With recent refractoring done around bigquery, you should not be able to cache results of get_core_table_details easily.